build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -64,10 +65,12 @@ Future<ServeHandler> watch(
   var terminator = new Terminator(terminateEventStream);
 
   overrideBuildConfig ??= await findBuildConfigOverrides(options.packageGraph);
-  final buildActions = await createBuildActions(options.packageGraph, builders,
+  final targetGraph = await TargetGraph.forPackageGraph(options.packageGraph);
```

Comment:
Should we add `targetGraph` to `options` so we don't have to recompute this?

Reply:
I could. I'd either need to change all the file IO to sync so we can synchronously create the TargetGraph along with the options, or change the way we get options to a static async method.